lib/vfscore/vfs.h

```diff
@@ -69,89 +69,760 @@ extern int vfs_debug;
-int  sys_futimens(int fd, const struct timespec times[2]);
-int  sys_fallocate(struct vfscore_file *fp, int mode, loff_t offset, loff_t len);
 
-int	 sys_pivot_root(const char *new_root, const char *old_put);
-void	 sync(void);
-int	 sys_chmod(const char *path, mode_t mode);
-int	 sys_fchmod(int fd, mode_t mode);
+/**
+ * Same as sys_utimensat, but the file to be updated is specified via
+ * an open file descriptor fd.
+ *
+ * @param fd
+ *	An open file descriptor pointing to the file whose timestamp would
+ *	be updated.
+ * @param times
+ *	The structure containing the new imestamps
+ * @return
+ *	0 - on success, error code on failure
+ */
+int sys_futimens(int fd, const struct timespec times[2]);
 
+/**
+ * Manipulates the file space for the file referred by fd, starting
+ * at offset and continuing for len bytes.
+ *
+ * @param fp
+ *	Pointer to the vfscore_file structure
+ * @param mode
+ *	Specifies how to manipulates the space. The mode can have the
+ *	the following values:
+ *		* 0 - allocates len bytes starting from offset.
+ *		* FALLOC_FL_KEEP_SIZE - the file change will not change even
+ *			if offset + len is greater than file size.
+ *		* FALLOC_FL_PUNCH_HOLE - deallocates len bytes starting at
+ *			offset and creates a hole in the file. This flag should
+ *			always come together with FALLOC_FL_KEEP_SIZE, otherwise
+ *			the sys_futimens call will fail.
+ * @param offset
+ *	Starting byte from which this operation would apply
+ * @param len
+ *	Number of bytes for which this operation would apply
+ * @return
+ *	0 - on success, error code on failure
+ */
+int sys_fallocate(struct vfscore_file *fp, int mode, loff_t offset, loff_t len);
 
-int	 task_alloc(struct task **pt);
-int	 task_conv(struct task *t, const char *path, int mode, char *full);
-int	 path_conv(char *wd, const char *cpath, char *full);
+/**
+ * This function is not used at this point.
+ */
+int sys_pivot_root(const char *new_root, const char *old_put);
 
-//int	 sec_file_permission(task_t task, char *path, int mode);
-int	 sec_vnode_permission(char *path);
+/**
+ * Calls the wrapper Unikraft sync syscall function.
+ *
+ */
+void sync(void);
 
-int     namei(const char *path, struct dentry **dpp);
-int	 namei_last_nofollow(char *path, struct dentry *ddp, struct dentry **dp);
-int	 lookup(char *path, struct dentry **dpp, char **name);
-void	 vnode_init(void);
-void	 lookup_init(void);
+/**
+ * Changes mode of the file whose pathname is given in pathname.
+ *
+ * @param path
+ *	Path to the file
+ * @param mode
+ *	The new file mode is specified in mode.
+ * @return
+ *	0 - on success, error code on failure
+ * @return
+ */
+int	sys_chmod(const char *path, mode_t mode);
 
-int     vfs_findroot(const char *path, struct mount **mp, char **root);
-int	 vfs_dname_copy(char *dest, const char *src, size_t size);
+/**
+ * Changes mode of the file specified to by the open file descriptor fd.
+ *
+ * @param fd
+ *	An open file descriptor.
```

Comment:
```suggestion
 *	An open file descriptor
```
Only add dots at the end of complete sentences.